src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.ValidateToken.Internal.cs

```diff
@@ -341,7 +341,7 @@ await ValidateJWSAsync(actorToken, actorParameters, configuration, callContext,
 
             ValidationResult<ValidatedSigningKeyLifetime> issuerSigningKeyValidationResult =
                 validationParameters.IssuerSigningKeyValidator(
-                    signatureValidationResult.UnwrapResult(), jsonWebToken, validationParameters, configuration, callContext);
+                    jsonWebToken.SigningKey, jsonWebToken, validationParameters, configuration, callContext);
```

Comment:
is this the right return type?

Reply:
The `ValidationResult<ValidatedSigningKeyLifetime>` is unchanged, and it is what we are currently returning.

The change here is to take the `SecurityKey` from the `JsonWebToken` instead of from the signature validation result, as we did in the initial implementation.

Reply:
Using the key from the `JsonWebToken` is in line with the original behaviour in the now legacy path.